cannot build the object library (COOLPROP_OBJECT_LIBRARY)

### Description

The option of COOLPROP_OBJECT_LIBRARY doesn't work; this option is necessary to build libExternalMedia.a or its DLL.

### Steps to Reproduce

1. mkdir build; cd build
2. cmake .. -D COOLPROP_OBJECT_LIBRARY=ON

**Expected behavior:** making object files as supposed.

**Actual behavior:** error in configuration as below.
` cmake .. -DCOOLPROP_OBJECT_LIBRARY=ON
-- COOLPROP_INSTALL_PREFIX=/home/spinhalf/Work/ExternalMedia/externals/CoolProp.git/install_root
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- CoolProp version: 6.4.2dev
-- Found PythonInterp: /usr/bin/python2.7 (found suitable version "2.7.17", minimum required is "2.7") 
-- Found LibDL: /usr/lib/x86_64-linux-gnu/libdl.so  
-- Looking for C++ include memory
-- Looking for C++ include memory - found
-- Performing Test HAVE_SHARED_PTR_IN_STD_NAMESPACE
-- Performing Test HAVE_SHARED_PTR_IN_STD_NAMESPACE - Success
-- Found shared_ptr in std namespace using <memory> header.
CMake Error at CMakeLists.txt:528 (target_link_libraries):
  Object library target "CoolProp" may not link to anything.


-- Configuring incomplete, errors occurred!
See also "/home/spinhalf/Work/ExternalMedia/externals/CoolProp.git/build/CMakeFiles/CMakeOutput.log".
`

### Versions

**CoolProp Version:** 6.4.2  
**Operating System and Version:** Ubuntu 18.04.5 with gcc 7.5.0 (x86_64-linux-gnu)
**Access Method:** ExternalMedia library in OpenModelica

### Additional Information

A recommend patch of CMakeLists.txt is attached here.
[CMakeLists.patch.txt](https://github.com/CoolProp/CoolProp/files/6193924/CMakeLists.patch.txt)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -523,8 +523,10 @@ IF ( COOLPROP_OBJECT_LIBRARY OR COOLPROP_STATIC_LIBRARY OR COOLPROP_SHARED_LIBRA
   ELSE()
     MESSAGE(FATAL_ERROR "You have to build a static or shared library.")
   ENDIF()
-
-  target_link_libraries (${LIB_NAME} ${CMAKE_DL_LIBS})
+ 
+  if (NOT COOLPROP_OBJECT_LIBRARY)
+    target_link_libraries (${LIB_NAME} ${CMAKE_DL_LIBS})
+  endif()
 
   # For windows systems, bug workaround for Eigen
   IF (MSVC90)

